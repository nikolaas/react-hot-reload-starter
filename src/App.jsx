import React from 'react';
import { Switch, Route } from 'react-router';
import './styles/index.scss';
import { asyncComponent } from './components/AsyncComponent';
import Main from './routes/Main';
import NotFound from './routes/NotFound';

const GetStarted = asyncComponent(() => import('./routes/GetStarted'));
const SyncCounter = asyncComponent(() => import('./routes/SyncCounter'));
const AsyncComputer = asyncComponent(() => import('./routes/AsyncComputer'));

export default () => {
    return (
        <Switch>
            <Route exact path="/" component={Main}/>
            <Route exact path="/get-started" component={GetStarted}/>
            <Route exact path="/sync-counter" component={SyncCounter}/>
            <Route exact path="/async-computer" component={AsyncComputer}/>
            <Route component={NotFound}/>
        </Switch>
    );
};
