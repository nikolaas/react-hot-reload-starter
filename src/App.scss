@import './theme/fonts';

.App {
  font-size: $font-size;
}
