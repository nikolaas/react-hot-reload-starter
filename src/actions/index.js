import { increment } from './counter-actions';

export { increment };
