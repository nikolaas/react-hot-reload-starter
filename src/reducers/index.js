import { combineReducers } from 'redux';
import { counterReducer } from './counter-reducer';

const asyncReducers = {};
export const injectReducer = (name, reducer) => {
    if (!reducer) {
        throw new Error('reducer must be function');
    }
    if (asyncReducers[name] === reducer) {
        return;
    }
    asyncReducers[name] = reducer;
};

export const createReducer = () => combineReducers({
    ...asyncReducers,
    counter: counterReducer,
});
