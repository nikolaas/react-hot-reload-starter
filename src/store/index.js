import { configureStore, registerAsyncReducer } from './store';

export { configureStore, registerAsyncReducer };
