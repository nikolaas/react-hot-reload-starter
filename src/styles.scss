@import '~normalize.css';
@import './theme/index';

body {
    font-size: themed('font.size');
}