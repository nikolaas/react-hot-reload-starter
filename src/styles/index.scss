@import '~normalize.css';
@import '../theme/index';
@import './grid';

html, body, #root {
    height: 1px;
    min-height: 100%;
}

body {
    font-size: themed('font.size');
}

ul {
    margin: 0;
    padding: 0;
    list-style-type: none;
}
