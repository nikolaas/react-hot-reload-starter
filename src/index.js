import React from 'react';
import ReactDOM from 'react-dom';
import { AppContainer } from 'react-hot-loader';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { configureStore } from './store';
import { createReducer } from './reducers';
import App from './App';

const store = configureStore(createReducer());

const render = Component => {
    ReactDOM.render(
        <AppContainer>
            <BrowserRouter basename={process.env.APP_BASE_URL}>
                <Provider store={store}>
                    <Component/>
                </Provider>
            </BrowserRouter>
        </AppContainer>,
        document.getElementById('root'),
    );
};

render(App);

if (module.hot) {
    // Hot reload application
    module.hot.accept('./App', () => {
        render(App);
    });

    // Hot reload reducers
    module.hot.accept('./reducers', () => {
        store.replaceReducer(createReducer());
    });

    //TODO reload store on update "./store" module
}
