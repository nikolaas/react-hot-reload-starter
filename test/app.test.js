import React from 'react';
import { shallow } from 'enzyme';
import { RawApp } from '../src/App';

describe('App tests', () => {
    it('should render the app', () => {
        const wrapper = shallow(<RawApp onIncrement={() => {}}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should trigger onIncrementEvent by click on increment button', () => {
        const incrementSpy = jest.fn();
        const wrapper = shallow(<RawApp onIncrement={incrementSpy}/>);
        wrapper.find('button').simulate('click');
        expect(incrementSpy).toHaveBeenCalled();
    });
});
