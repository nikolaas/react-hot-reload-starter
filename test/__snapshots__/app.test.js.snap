// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should render the app 1`] = `
<div
  className="App"
>
  <h1>
    Counter 
  </h1>
  <p>
    Counter: 
    <span
      style={
        Object {
          "color": "#f00",
        }
      }
    >
      0
    </span>
  </p>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
</div>
`;
